app/models/user.rb

```diff
@@ -49,71 +49,92 @@ def should_generate_new_friendly_id?
   source: :movie
 
   def all_lists
-    (self.lists | self.member_lists).uniq
+    list_ids = lists.pluck(:id)
+    member_list_ids = member_lists.pluck(:id)
+    ids = (list_ids + member_list_ids).uniq
+    List.where(id: ids)
   end
 
   def all_listings
-    (self.listings | self.member_listings)
+    listing_ids = listings.pluck(:id)
+    member_listing_ids = member_listings.pluck(:id)
+    ids = (listing_ids + member_listing_ids).uniq
+    Listing.where(id: ids)
   end
 
   def lists_except_movie(movie = nil)
-    if movie.present? && movie.in_db
-      (all_lists - movie.lists.by_user(self))
-        .sort_by { |list| list.name.downcase }
-    else
-      all_lists_by_name
-    end
+    return all_lists_by_name unless movie&.in_db
+    movie_list_ids = movie.lists.by_user(self).pluck(:id)
+    all_lists.where.not(id: movie_list_ids)
   end
 
   def all_lists_by_name
-    self.all_lists.sort_by { |list| list.name.downcase }
+    # TODO: not sure what this is acting weird
+    # all_lists.order('lower(name) ASC')
+    all_lists.sort_by { |list| list.name.downcase }
```

Comment:
This one I did not update to use sql because the results were fucky. I need to look more into it, but for now, I just kept it as-is.